src/test/python/test_imsc_reader.py

```diff
@@ -35,9 +35,52 @@
 import ttconv.model as model
 import ttconv.style_properties as styles
 import ttconv.imsc.reader as imsc_reader
+import ttconv.imsc.namespaces as xml_ns
 
 class IMSCReaderTest(unittest.TestCase):
 
+  def setUp(self):
+    et.register_namespace("ttml", xml_ns.TTML)
+    et.register_namespace("ttp", xml_ns.TTP)
+    et.register_namespace("tts", xml_ns.TTS)
+    et.register_namespace("ittp", xml_ns.ITTP)
+    et.register_namespace("itts", xml_ns.ITTS)
+
+  def test_reader_tt_element_not_root_element(self):
+
+    xml_str = """<?xml version="1.0" encoding="UTF-8"?>
+      <not_tt xml:lang="en"
+          xmlns="http://www.w3.org/ns/ttml"
+          xmlns:ttm="http://www.w3.org/ns/ttml#metadata" 
+          xmlns:tts="http://www.w3.org/ns/ttml#styling"
+          xmlns:ttp="http://www.w3.org/ns/ttml#parameter" 
+          xmlns:ittp="http://www.w3.org/ns/ttml/profile/imsc1#parameter"
+          ittp:activeArea="50% 50% 80% 80%"
+          tts:extent="640px 480px"
+          ttp:profile="http://www.w3.org/ns/ttml/profile/imsc1/text">
+      </not_tt>"""
+
+    tt_not_root = et.ElementTree(et.fromstring(xml_str))
+    self.assertIsNone(imsc_reader.to_model(tt_not_root))
+
+  def test_reader_invalid_tt_element(self):
+
+    xml_str = """<?xml version="1.0" encoding="UTF-8"?>
+      <tt xml:lang="en"
+          xmlns="http://www.w3.org/ns/ttml"
+          xmlns:ttm="http://www.w3.org/ns/ttml#metadata" 
+          xmlns:tts="http://www.w3.org/ns/ttml#styling"
+          xmlns:ttp="http://www.w3.org/ns/ttml#parameter" 
+          xmlns:ittp="http://www.w3.org/ns/ttml/profile/imsc1#parameter"
+          ittp:activeArea="50% 50% 80% 80%"
+          tts:extent="640px 480px"
+          ttp:profile="http://www.w3.org/ns/ttml/profile/imsc1/text">
+          <asdf></asdf>
+      </tt>"""
+
```

Comment:
This XML fails as <asdf> is not a valid TT value. The reader throws. 
Is the second None return in reader even valid?
What XML would trigger that case?
It seems the reader throws in many cases of invalid XML.

Reply:
Replace `<tt>` with `<foo>`, i.e. `None` will be returned if the root element is not `<tt>`

Reply:
Oh... this is already covered by `test_reader_tt_element_not_root_element`. I suggest removing this test.

Reply:
yes, that is correct. I wasn't sure if there was another way to pass the first check in to_model while still triggering the second check. I tried several things but the to_model alway threw exceptions which is the right way to do it.